file_read_backwards/file_read_backwards.py

```diff
@@ -40,34 +38,32 @@ def __init__(self, path, encoding="utf-8", chunk_size=io.DEFAULT_BUFFER_SIZE):
         self.path = path
         self.encoding = encoding.lower()
         self.chunk_size = chunk_size
-        self.__fp_refs = []
+        self.iterator = FileReadBackwardsIterator(io.open(self.path, mode="rb"), self.encoding, self.chunk_size)
 
     def __iter__(self):
-        """Opens a file and returns a reversed iterator on it."""
-        fp = io.open(self.path, mode="rb")
-        self.__fp_refs.append(weakref.ref(fp))
-        return FileReadBackwardsIterator(fp, self.encoding, self.chunk_size)
+        """Return its iterator."""
+        return self.iterator
 
     def __enter__(self):
         return self
 
     def __exit__(self, exc_type, exc_val, exc_tb):
-        """Closes all opened file handlers and propagates all exceptions on exit."""
+        """Closes all opened its file handler and propagates all exceptions on exit."""
         self.close()
         return False
 
     def close(self):
-        """Closes all opened file handlers.
+        """Closes all opened it s file handler."""
+        self.iterator.close()
```

Comment:
Not forgetting, if `self.iterator` becomes lazy to check for `None`

Reply:
:-) thanks for the reminder